src/legendre_fet.cpp

```diff
@@ -0,0 +1,308 @@
+                         std::shared_ptr<EnergyFilter> energy_in,
+                         std::vector<LegendreFET::Axis> axes, size_t FET_order,
+                         LegendreFET::Quantity quantity_,
+                         LegendreFET::Estimator estimator_, std::string name_)
+    : ITally(quantity_, estimator_, name_),
+      cartesian_filter_(position_filter_),
+      energy_in_(energy_in),
+      axes_(),
+      FET_order_(FET_order) {
+  // get the shape or dimensions for the cartesian_filter_
+  // the Legendre-FET requires the co-ordinate information
+  StaticVector3 position_shape;
+  if (cartesian_filter_ == nullptr) {
+    fatal_error("For the tally, " + name_ +
+                ", the positional filter is required to do the legendre "
+                "functional expansion tally.");
+  }
+  position_shape = position_filter_->get_shape();
+
+  StaticVector6 tally_dimensions_(position_shape.begin(), position_shape.end());
+
+  // add the dimension for energy_in_ only if exist
+  if (energy_in_) {
+    std::size_t ne = energy_in_->size();
+    tally_dimensions_.insert(tally_dimensions_.begin(), ne);
+  }
+
+  // for legendre, check the size of axis vector should be between than 3.
+  for (auto& c : axes) {
+    axes_.push_back(c);
+  }
+  if (axes_.size() <= 1 && axes_.size() >= 3)
+    fatal_error(
+        "The no. of given axis for legendre-FET is not between 1 to 3 for "
+        "tally " + name_ + ".");
+
+  // Add the dimension for axis only when greater than 1
+  std::size_t n_axis = axes_.size();
+  tally_dimensions_.push_back(n_axis);
+
+  // Add the dimension for the order of FET only if greater than 0
+  std::size_t fet_order_dimension = FET_order_ + 1;
+  tally_dimensions_.push_back(fet_order_dimension);
+
+  // reallocate and fill with zeros for the tally avg, gen-score and variance
+  tally_avg.reallocate(tally_dimensions_);
+  tally_avg.fill(0.0);
+
+  tally_gen_score.reallocate(tally_dimensions_);
+  tally_gen_score.fill(0.0);
+
+  tally_var.reallocate(tally_dimensions_);
+  tally_var.fill(0.0);
+}
+
+void LegendreFET::score_collision(const Particle& p, const Tracker& tktr,
+                                  MaterialHelper& mat) {
+  // get the cartisian_filter indices
+  StaticVector3 position_index = cartesian_filter_->get_indices(tktr);
+  if (position_index.empty()) {
+    // No bin is found, don't score.
+    return;
+  }
+  StaticVector6 indices(position_index.begin(), position_index.end());
+
+  // get the energy-index, if energy-filter exists
+  if (energy_in_) {
+    std::optional<std::size_t> E_indx = energy_in_->get_index(p.E());
+
+    if (E_indx.has_value()) {
+      std::size_t index_E = E_indx.value();
+      indices.insert(indices.begin(), index_E);
+    } else
+      // Not inside any energy bin. Don't score.
+      return;
+  }
+
+  const double Et = mat.Et(p.E());
+
+  const double collision_score = particle_base_score(p, mat) / Et;
```

Comment:
Isn't this resulting in a double division by `Et` ? Because `particle_base_score` will already return `w/Et` if we have a collision estimator for the flux.